test/e2e/prometheus_test.go

```diff
@@ -117,26 +117,25 @@ func TestPrometheusAlertmanagerAntiAffinity(t *testing.T) {
 	}
 
 	var alm = `affinity:
-	   podAntiAffinity:
-	      preferredDuringSchedulingIgnoredDuringExecution:
-	      - podAffinityTerm:
-	          labelSelector:
-	            matchExpressions:
-	            - key: alertmanager
-	              operator: In
-	              values:
-	              - main`
+           podAntiAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
+              - podAffinityTerm:
+                  labelSelector:
+                    matchLabels:
+                      app.kubernetes.io/component: alert-router
+                      app.kubernetes.io/name: alertmanager
+                      app.kubernetes.io/part-of: openshift-monitoring`
 
 	var k8s = `affinity:
-	   podAntiAffinity:
-	      preferredDuringSchedulingIgnoredDuringExecution:
-	      - podAffinityTerm:
-	          labelSelector:
-	            matchExpressions:
-	            - key: prometheus
-	              operator: In
-	              values:
-	              - k8s`
+           podAntiAffinity:
+              preferredDuringSchedulingIgnoredDuringExecution:
```

Comment:
```suggestion
              requiredDuringSchedulingIgnoredDuringExecution:
```

Reply:
note that this will require to update the ranges that are used to parse anti-affinity

Reply:
like I said in a previous comment, it should probably check that `len(pod.Spec.Affinity.PodAntiAffinity.RequiredDuringSchedulingIgnoredDuringExecution) > 0` instead of trying to parse the stringified representation of the field :)

Reply:
ty, nice catch.

Reply:
yes definitely, but for now let's fix the expected output to proceed with this PR.